drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -105,23 +105,27 @@ int intel_adsp_hda_dma_link_in_config(const struct device *dev,
 {
 	const struct intel_adsp_hda_dma_cfg *const cfg = dev->config;
 	struct dma_block_config *blk_cfg;
+	uint32_t size = 0;
 	uint8_t *buf;
 	int res;
 
 	__ASSERT(channel < cfg->dma_channels, "Channel does not exist");
-	__ASSERT(dma_cfg->block_count == 1,
-		 "HDA does not support scatter gather or chained "
-		 "block transfers.");
 	__ASSERT(dma_cfg->channel_direction == cfg->direction,
 		 "Unexpected channel direction, HDA link in supports "
 		 "PERIPHERAL_TO_MEMORY");
 
 	blk_cfg = dma_cfg->head_block;
-	buf = (uint8_t *)(uintptr_t)(blk_cfg->source_address);
+	buf = (uint8_t *)(uintptr_t)(blk_cfg->dest_address);
+
+	for (int i = 0; i < dma_cfg->block_count; i++) {
```

Comment:
Is the assumption that the blocks are contiguously allocated but treated as separate buffers? That might be ok, but should be validated with an __ASSERT